pkg/store/proxy.go

```diff
@@ -274,6 +277,80 @@ func (s *ProxyStore) TSDBInfos() []infopb.TSDBInfo {
 	return infos
 }
 
+func (s *ProxyStore) LabelNamesBloom() bloom.Filter {
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+
+	return s.labelNamesBloom
+}
+
+func (s *ProxyStore) UpdateLabelNamesBloom(ctx context.Context) error {
```

Comment:
Can the proxy store do a union of all bloom filters?

Reply:
So I initially thought this might be a good way, where proxy store can just call LabelNamesBloom on each store, and do a union, but merging bloom filters is difficult given their nature.
The number of items and number of hashing functions need to be the same, for two bloom filters to be mergeable https://github.com/bits-and-blooms/bloom/blob/master/bloom.go#L173

If we want to make that constant, than false positive rates might vary. So I choose to do a union of labelnames instead.

Reply:
That makes sense 👍